bofire/utils/multiobjective.py

```diff
@@ -134,26 +171,76 @@ def infer_ref_point(
     domain: Domain,
     experiments: pd.DataFrame,
     return_masked: bool = False,
+    reference_point: Optional[ExplicitReferencePoint] = None,
 ) -> Dict[str, float]:
+    """Method to infer the reference point for a given domain and experiments.
+
+    Args:
+        domain: Domain for which the reference point should be inferred.
+        experiments: Experiments for which the reference point should be inferred.
+        return_masked: If True, the masked reference point is returned. If False,
+            the unmasked reference point is returned. Defaults to False.
+        reference_point: Reference point to be used. If None is provided, the
+            reference value is inferred by the worst values seen so far for
+            every objective. Defaults to None.
+    """
     outputs = domain.outputs.get_by_objective(
         includes=[MaximizeObjective, MinimizeObjective, CloseToTargetObjective],
     )
-    keys = [f.key for f in outputs]
+    keys = outputs.get_keys()
+
+    if reference_point is None:
+        reference_point = ExplicitReferencePoint(
+            values={
+                key: AbsoluteMovingReferenceValue(orient_at_best=False, offset=0.0)
+                for key in keys
+            }
+        )
+
     objective = get_multiobjective_objective(outputs=outputs, experiments=experiments)
 
     df = domain.outputs.preprocess_experiments_all_valid_outputs(
         experiments,
         output_feature_keys=keys,
     )
 
-    ref_point_array = (
+    worst_values_array = (
         objective(torch.from_numpy(df[keys].values).to(**tkwargs), None)
         .numpy()
         .min(axis=0)
     )
 
+    best_values_array = (
+        objective(torch.from_numpy(df[keys].values).to(**tkwargs), None)
+        .numpy()
+        .max(axis=0)
+    )
+
+    # in the ref_point_array we have the masked values, which means that
```

Comment:
Can you please re-formulate this comment. I don't understand what it means. What do masked values have to do with maximization?